quipucords/tests/integration/test_openshift_scan.py

```diff
@@ -214,7 +227,13 @@ def scan_response(self, django_client, scan_id, scan_manager):
         """Start a scan job and poll its results endpoint until completion."""
         create_scan_job_response = django_client.post(f"scans/{scan_id}/jobs/")
         assert create_scan_job_response.ok, create_scan_job_response.text
-        scan_manager.work()
+        # Scan manager's work() attempts to fetch data from openshift, for which
+        # we Mock the OpenShiftApi. Note that Mocks do not work on MacOS as the
+        # default invocation method is spawn instead of a fork. If using spawn,
+        # a new Python interpreter is launched in which the Mocks are not
+        # honored so we need to invoke the tested code via fork.
+        with set_mp_start_method("fork"):
+            scan_manager.work()
         response = django_client.get(f"scans/{scan_id}/")
```

Comment:
NICE JOB! I love this solution (and especially the explanation) surrounding it.
With that said, consider this totally optional suggestion:

Since `scan_manager` will always spawn new processes under mac we might do this mock under `work` method.
This would be the relevant code: https://github.com/quipucords/quipucords/blob/5a0bd9a8d7720a1d08c705c66517872d6515cae5/quipucords/tests/dummy_scan_manager.py#L48-L59

My suggestion here is to call `current_job.start()` inside the modified multiprocess context.

Reply:
👍 I'll give it a try.